precompiles/erc20/query.go

```diff
@@ -20,3 +36,199 @@ const (
 	// query.
 	BalanceOfMethod = "balanceOf"
 )
+
+// Name returns the name of the token. If the token metadata is registered in the
+// bank module, it returns its name. Otherwise it returns the base denomination of
+// the token capitalized (eg. uatom -> Atom).
+func (p Precompile) Name(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	_ vm.StateDB,
+	method *abi.Method,
+	_ []interface{},
+) ([]byte, error) {
+	metadata, found := p.bankKeeper.GetDenomMetaData(ctx, p.tokenPair.Denom)
+	if found {
+		return method.Outputs.Pack(metadata.Name)
+	}
+
+	// Infer the denomination name from the coin denomination base denom
+	denomTrace, err := GetDenomTrace(p.transferKeeper, ctx, p.tokenPair.Denom)
+	if err != nil {
+		// FIXME: return not supported (same error as when you call the method on an ERC20.sol)
```

Comment:
this might be clearer what is meant
```suggestion
		// FIXME: return 'not supported' (same error as when you call the method on an ERC20.sol)
```